rules/downgrade/src/Rector/Property/TypedPropertyRector.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Downgrade\Rector\Property;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\UnionType;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\Downgrade\Tests\Rector\Property\TypedPropertyRector\TypedPropertyRectorTest
+ */
+final class TypedPropertyRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Changes property type definition from type definitions to `@var` annotations.', [
+            new CodeSample(
+                <<<'PHP'
+class SomeClass {
+    private string $property;
+}
+PHP
+,
+                <<<'PHP'
+class SomeClass {
+    /**
+    * @var string
+    */
+    private $property;
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->type === null) {
+            return null;
+        }
+
+        $propertyTypeDefinition = $this->getPropertyTypeDefinition($node->type);
+        if ($propertyTypeDefinition === null) {
+            return null;
+        }
+
+        $docComment = $node->getDocComment();
+        if ($docComment !== null) {
+            $node->type = null;
+            return $node;
+        }
+
+        $node->setDocComment(new Doc(sprintf("/**\r\n* @var %s\r\n*/", $propertyTypeDefinition)));
```

Comment:
Actually, Rector uses own object for doc operations.

You can fidn example here: https://github.com/rectorphp/rector/blob/05d518bce33e60cdd7ebec31e01e1b701b30f4d1/rules/solid/src/Rector/ClassMethod/ChangeReadOnlyVariableWithDefaultValueToConstantRector.php#L215-L221

Reply:
Yeah, yeah, I saw... playing with it at the moment.